ports/espressif/common-hal/espcamera/Camera.c

```diff
@@ -77,13 +77,12 @@ void common_hal_espcamera_camera_construct(
 
     self->i2c = i2c;
 
-    self->camera_config.pin_pwdn = common_hal_mcu_pin_number(powerdown_pin);
-    self->camera_config.pin_reset = common_hal_mcu_pin_number(reset_pin);
-    self->camera_config.pin_xclk = common_hal_mcu_pin_number(external_clock_pin);
+    // These pins might be NULL because they were not specified.
+    self->camera_config.pin_pwdn = powerdown_pin ? common_hal_mcu_pin_number(powerdown_pin) : NO_PIN;
```

Comment:
common_hal_mcu_pin_number is implemented thus on espressif:
```
uint8_t common_hal_mcu_pin_number(const mcu_pin_obj_t *pin) {
    return pin ? pin->number : NO_PIN;
}
```
that is, the special NO_PIN value is returned for a NULL pointer input. So, I don't know why this change would be needed.

Reply:
It's because the fields in `camera_config` are ints. `NO_PIN` is defined as `-1`, and the camera driver checks for the pin being `>= 0`. _However_, `common_hal_mcu_pin_number()` returns a `uint8_t`, so the `-1` turns into `255`, and that makes it still `>= 0`. I will put in a explicating comment.

Reply:
I noticed that there were GPIO errors being printed in the debug log -- that's why I fixed this.